### PR TITLE
Gen2: add MonoCamera and StereoDepth nodes

### PR DESCRIPTION
**Pull request by alex-luxonis**
Related PR: https://github.com/luxonis/depthai-shared/pull/10

### CODE REVIEW

**Review by themarpe — CHANGES_REQUESTED** [review]
LGTM apart from checks failing.
For style, run the `clangformat` target to modify style according to clang format style file (warning, it modifies in place, make sure to commit before, in case something goes wrong)
```
cmake --build [path/to/build/dir] --target clangformat
```

File: include/depthai/pipeline/node/StereoDepth.hpp [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    std::vector<Output> getOutputs() override;
+    std::vector<Input> getInputs() override;
+    nlohmann::json getProperties() override;
+    std::shared_ptr<Node> clone() override;
+
+  public:
+    StereoDepth(const std::shared_ptr<PipelineImpl>& par, int64_t nodeId);
+
+    Input left{*this, "left", Input::Type::SReceiver, {{DatatypeEnum::Buffer, true}}};
+    Input right{*this, "right", Input::Type::SReceiver, {{DatatypeEnum::Buffer, true}}};
```
Comment (themarpe) [inline include/depthai/pipeline/node/StereoDepth.hpp:24]: This is fixed now in gen2_imagemanip. Check method `canConnect`. Preferably set back to ImgFrame and apply the fix from gen2_imagemanip to `canConnect`

**Review by themarpe — CHANGES_REQUESTED** [review]

File: include/depthai/pipeline/node/StereoDepth.hpp (hunk 1 above)
Comment (alex-luxonis) [reply include/depthai/pipeline/node/StereoDepth.hpp]: Done. Thanks!

**Comment by themarpe** [comment]
Merged to `gen2_develop`

**Comment by alex-luxonis** [comment]
Reopening to change target branch, to get marked as "merged".

**Pull request merged**